### DISCUSSION TITLE
Montage Page formatting help.

### DISCUSSION THREAD

**Discussion by blaughlin1992** [ImageMagick/ImageMagick#7279] [Consulting]
Will probably take you 30 seconds. I need help formatting a montage page.  I am close.  Just need:
1 - wider margins around the edge.  right marging is my problem.
2 - captions in or under each image with the file name minus the extension.

Here is what I have: 	magick montage *.{heic,jpg} -tile 3x5 -geometry +50+50 -background "#FFFFFF" -quality 85 -density 150x150 -page letter+0+5 "C:\Users\blaughlin\Photos_to_Batch\Processed\!newname!_Photos.pdf"

**Comment by fmw42** [comment]
It would help to have an image or for you to show your output.  If you want to add to the margin on one side, post process with -splice.  Pipe the output of montage to magick and do the splice.  See https://imagemagick.org/Usage/crop/#splice. To add a caption below each image in montage, see https://imagemagick.org/Usage/montage/#label.  Use %t in place of %f to get just the filename without the suffix.

**Reply by blaughlin1992** [reply]
I have been trying this but get errors whenever I use the -label command.  Wondering if it is an install or operating system issue

**Comment by blaughlin1992** [comment]
Thank you!  Here is the best I could do.
[TEST JPG_Photos.pdf](https://github.com/ImageMagick/ImageMagick/files/15168633/TEST.JPG_Photos.pdf)
Yes, I tried those label commands, but got nothing.

**Comment by blaughlin1992** [comment]
Can you revise my line of code to what you think would work?

**Comment by fmw42** [comment]
Try this if you want to add some white border on both the left and right sides to make it similar to the top and bottom.  Change "X" to whatever number of pixels on each of left and right sides that you want to add and 0 on the top and bottom.  Notice the -label "%t"

`magick montage *.{heic,jpg} -label "%t" -tile 3x5 -geometry +50+50 -background "#FFFFFF" -quality 85 -density 150x150 -page letter+0+5 miff:- | magick - -bordercolor white -border Xx0 "C:\Users\blaughlin\Photos_to_Batch\Processed!newname!_Photos.pdf"`

See https://imagemagick.org/Usage/crop/#border for -border
See https://imagemagick.org/Usage/montage/#label for adding titles (via -label) to each images

**Comment by fmw42** [comment]
My mistake. Put the -label "%t" before your input images.

`magick montage -label "%t" *.{heic,jpg} -tile 3x5 -geometry +50+50 -background "#FFFFFF" -quality 85 -density 150x150 -page letter+0+5 miff:- | magick - -bordercolor white -border Xx0 "C:\Users\blaughlin\Photos_to_Batch\Processed!newname!_Photos.pdf"`

**Comment by blaughlin1992** [comment]
Fred!  Thank you for your help!  I think I have gotten it to the best place I can.  Here is the script.  It also moves and renames some files.  But it ended up creating annotated images and then adding them to the montage.  The HEIC formatted images also had to be converted to jpg for it to work:
[Ardrey Kell HS HEICs_Photos.pdf](https://github.com/ImageMagick/ImageMagick/files/15177529/Ardrey.Kell.HS.HEICs_Photos.pdf)


[batch_contact_sheet_by_WBL NEW6 - Copy.txt](https://github.com/ImageMagick/ImageMagick/files/15177491/batch_contact_sheet_by_WBL.NEW6.-.Copy.txt)

**Comment by fmw42** [comment]
Why does HEIC images fail?  How?  Can you provide two or three HEIC images so that I can check that out?

Thanks for the code, but I do not work on a Windows platform and do not code in Windows batch. I understand the Imagemagick code that you used.
